### PR TITLE
docs: update to include MMDS V2 behavior

### PR DESCRIPTION
**Pull request by luminitavoicu**
Signed-off-by: Luminita Voicu <lumivo@amazon.com>

# Reason for This PR

Include documentation describing MMDS v2 behavior.
This PR is work in progress because it lacks specification regarding MMDS default behavior and version configuration.

- [ ] This functionality can be added in [`rust-vmm`][1].

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The issue which led to this PR has a clear conclusion.
- [ ] This PR follows the solution outlined in the related issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes follow the [Runbook for Firecracker API changes][2].
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

[1]: https://github.com/rust-vmm
[2]: ../docs/api-change-runbook.md

### CODE REVIEW

File: docs/mmds/mmds-user-guide.md [hunk 1]
```diff
@@ -175,12 +175,66 @@ Output:
...
 
-To retrieve existing MMDS metadata from guest operating system, an HTTP `GET`
+Accessing the contents of the metadata store from guest operating system
+can be done using one of the following methods:
+- `V1`: simple request/response method
+- `V2`: session-oriented method
+
+#### Version 1
+
+To retrieve existing MMDS metadata using MMDS version 1, an HTTP `GET`
```
Comment (raduiliescu) [inline docs/mmds/mmds-user-guide.md:185]: I would add that MMDS v1 does not support PUT.

File: docs/mmds/mmds-user-guide.md [hunk 2]
```diff
@@ -276,6 +330,12 @@ The request was successfully processed and a response was successfully formed.
 
 The request was malformed.
 
+*401* - `Unauthorized`
```
Comment (raduiliescu) [inline docs/mmds/mmds-user-guide.md:333]: Should we also document the error case of mmdsv1 put?

**Review by raduiliescu — APPROVED** [review]

File: docs/mmds/mmds-user-guide.md (hunk 1 above)
Comment (luminitavoicu) [reply docs/mmds/mmds-user-guide.md:224]: Done

File: docs/mmds/mmds-user-guide.md (hunk 2 above)
Comment (luminitavoicu) [reply docs/mmds/mmds-user-guide.md:390]: Done

**Review by raduiliescu — APPROVED** [review]

**Review by AlexandruCihodaru — APPROVED** [review]

**Pull request merged**